syft/workers/plan.py

```diff
@@ -0,0 +1,261 @@
+from .base import BaseWorker
+from syft.codes import MSGTYPE
+import syft as sy
+import random
+
+
+class PlanPointer:
+    def __init__(self, id, location, id_at_location, owner):
+        self.id = id
+        self.location = location
+        self.id_at_location = id_at_location
+        self.owner = owner
+
+
+def replace_ids(obj, change_id, to_id, from_worker, to_worker):
+    _obj = list()
+
+    for i, item in enumerate(obj):
+        if isinstance(item, int) and (item == change_id):
+            _obj.append(to_id)
+
+        elif isinstance(item, type(from_worker)) and (item == from_worker):
+            _obj.append(to_worker)
+
+        elif isinstance(item, (list, tuple)):
+            _obj.append(
+                replace_ids(
+                    obj=item,
+                    change_id=change_id,
+                    to_id=to_id,
+                    from_worker=from_worker,
+                    to_worker=to_worker,
+                )
+            )
+
+        else:
+            _obj.append(item)
+
+    return _obj
+
+
+class Plan(BaseWorker):
```

Comment:
Is there a good reason that plans are a subclass of workers? What functionality do they inherit?

Reply:
They catch the call to function to build plans which is crucial to the plan functionality